### PR TITLE
Some important changes to the API

### PR DESCRIPTION
**Pull request by msyriac**
@qujia7 , I'd like your thoughts on these changes. I'm almost done integrating them into SO-Lenspipe through the rdn0 branch (attn: @NiallMac ). @ajvanengelen , this might be relevant for your ongoing work with Darby too.

The main changes/additions here are:

1. Cleaning up the way theory Cls are passed in (now through dictionaries, instead of an obscure orphics object)
2. Returning lensing potential instead of convergence (this is easiest in terms of matching conventions in the literature, Planck and Toshiya's norm code)
3. A set of utilities in falafel/utils.py that we can use in SO-lenspipe and elsewhere
4. Some cleanup of redundant function arguments, though I think more work is needed for shear and hardening
5. A better demo script in test/simple.py, which calls normalization from tempura

### CODE REVIEW

File: falafel/qe.py [hunk 1]
```diff
@@ -79,12 +79,18 @@ def map2alm_spin(self,imap,lmax,spin_alm,spin_transform):
             return cs.map2alm(enmap.enmap(dmap,imap.wcs),spin=spin_transform,lmax=lmax)
 
 
-def filter_alms(alms,ffunc,lmin=None,lmax=None):
+def filter_alms(alms,filt,lmin=None,lmax=None):
```
Comment (msyriac) [inline falafel/qe.py:82]: This function now needs 1d arrays, not lambda functions

**Comment by msyriac** [comment]
I noticed that functions like the source estimator effectively only work on one copy of the map. Doesn't this make it impossible to do RDN0 with it (which requires mixing sims and data)?  That might explain some of the weirdness we say with hardened RDN0. I'll push fixes for that here.

File: falafel/qe.py [hunk 2]
```diff
@@ -399,38 +388,25 @@ def qe_shear(px,mlmax,Talm=None,fTalm=None):
...
-
     if xfTalm is None:
-        if fTalm is not None: xfTalm = fTalm.copy()
-    if xfEalm is None:
-        if fEalm is not None: xfEalm = fEalm.copy()
-    if xfBalm is None:
-        if fBalm is not None: xfBalm = fBalm.copy()
+        xfTalm = fTalm.copy()
+    omap = enmap.zeros((2,)+px.shape,px.wcs) #load empty map with SO map wcs and shape
     rmap=px.alm2map_spin(np.stack((fTalm,fTalm)),0,0,ncomp=2,mlmax=mlmax)
```
Comment (msyriac) [inline falafel/qe.py:402]: Is this really needed? Isn't the source estimator just a simple spin 0
map2alm( alm2map(alm_inverse_filtered_1) * alm2map(alm_inverse_filtered_2) ) ?

Comment (qujia7) [reply falafel/qe.py:402]: That's right, I should have mentioned that I am working with the `bias_harden` branch and the two changes that you mentioned above were already there but I didn't update the pull request.

Comment (msyriac) [reply falafel/qe.py:402]: Oh cool! In the future, let's try to get things merged in (or at least made aware through a PR) as soon as possible.

**Comment by NiallMac** [comment]
All sounds great Mat. I've been trying to familiarize  myself (aka bugging @qujia7 :)) with the bias hardening stuff, so can have a go at updating that to fit with this overhaul once you've merged it in.

**Pull request merged**